### PR TITLE
RSpec 3

### PR DESCRIPTION
**Pull request by princemaple**
:koala: 

Blocked by #79 

/cc @zendesk/quokka

### CODE REVIEW

**Comment by atroche** [comment]
:+1:

File: lib/zendesk_apps_support/app_file.rb [hunk 1]
```diff
@@ -16,7 +16,7 @@ def read
     end
 
     def =~(regex)
-      @relative_path =~ regex
+      relative_path =~ regex
```
Comment (adammw) [inline lib/zendesk_apps_support/app_file.rb]: is this a bug/typo fixed or does `@relative_path` actually exist?

File: spec/app_file_spec.rb [hunk 2]
```diff
@@ -8,20 +8,20 @@
   end
 
   describe '=~' do
-    it 'should test against the relative path of the file' do
-      @file.should =~ /layout/
+    it 'tests against the relative path of the file' do
+      expect(@file =~ /layout/).to be_truthy
```
Comment (adammw) [inline spec/app_file_spec.rb]: why can't you use `expect(@file).to match(/layout/)` here as well?

File: spec/i18n_spec.rb [hunk 3]
```diff
@@ -15,7 +15,7 @@
     project_root = Pathname.new(File.expand_path('../../', __FILE__))
     zendesk_version  = project_root.join('config/locales/translations/zendesk_apps_support.yml')
     standard_version = project_root.join('config/locales/en.yml')
-    File.mtime(zendesk_version).to_i.should be <= File.mtime(standard_version).to_i
+    expect(File.mtime(zendesk_version).to_i).to be <= File.mtime(standard_version).to_i
```
Comment (adammw) [inline spec/i18n_spec.rb]: What is the `<=` doing here?

File: spec/package_spec.rb [hunk 4]
```diff
@@ -5,61 +5,61 @@
...
-    @package.stub(:lib_files) do |*args, &block|
+    allow(@package).to receive(:lib_files) do |*args, &block|
       lib_files_original_method.call(*args, &block).sort_by { |f| f.relative_path }
     end
   end
 
   describe 'files' do
     it 'should return all the files within the app folder excluding files in tmp folder' do
-      @package.files.map(&:relative_path).should =~ %w(app.css app.js assets/logo-small.png assets/logo.png lib/a.js lib/a.txt lib/nested/b.js manifest.json templates/layout.hdbs translations/en.json)
+      expect(@package.files.map(&:relative_path)).to match_array(%w(app.css app.js assets/logo-small.png assets/logo.png lib/a.js lib/a.txt lib/nested/b.js manifest.json templates/layout.hdbs translations/en.json))
```
Comment (adammw) [inline spec/package_spec.rb]: can you move this array to a separate line?

File: spec/package_spec.rb [hunk 5]
```diff
@@ -5,61 +5,61 @@
...
-      @package.lib_files.map(&:relative_path).should == %w(lib/a.js lib/nested/b.js)
+      expect(@package.lib_files.map(&:relative_path)).to eq(%w(lib/a.js lib/nested/b.js))
     end
   end
 
   describe 'commonjs_modules' do
     it 'should return an object with name value pairs containing the path and code' do
-      @package.commonjs_modules.should == {
+      expect(@package.commonjs_modules).to eq({
         "a.js"=>"var a = {\n  name: 'This is A'\n};\n\nmodule.exports = a;\n",
```
Comment (adammw) [inline spec/package_spec.rb]: spaces between `=>`?

File: spec/validations/validation_serialization_spec.rb [hunk 6]
```diff
@@ -54,12 +59,18 @@
...
+        subject { super().key }
+        it { is_expected.to eq(:jshint) }
+      end
 
-      its(:jshint_errors) do
-        should == [ { 'line' => 55, 'reason' => 'Yuck' } ]
+      describe '#jshint_errors' do
+        subject { super().jshint_errors }
+        it do
+        is_expected.to eq([ { 'line' => 55, 'reason' => 'Yuck' } ])
```
Comment (adammw) [inline spec/validations/validation_serialization_spec.rb]: indentation

**Comment by princemaple** [comment]
@adammw all style changes will be addressed in #81

File: lib/zendesk_apps_support/app_file.rb (hunk 1 above)
Comment (princemaple) [inline lib/zendesk_apps_support/app_file.rb]: There is a attr_reader in this class

File: spec/validations/manifest_spec.rb [hunk 7]
```diff
@@ -37,7 +37,7 @@ def create_package(parameter_hash)
   it 'should have an error when manifest.json is missing' do
     files = [double('AppFile', :relative_path => 'abc.json')]
     package = double('Package', :files => files)
-    package.should have_error 'Could not find manifest.json'
```
Comment (adammw) [inline spec/validations/manifest_spec.rb]: Is there a newer way of writing `have_error` now?

**Comment by adammw** [comment]
@princemaple could have told me earlier. :sweat_smile: 
:+1:

File: spec/app_file_spec.rb (hunk 2 above)
Comment (princemaple) [inline spec/app_file_spec.rb]: This I wonder too, will look into it.

File: spec/i18n_spec.rb (hunk 3 above)
Comment (sandlerr) [inline spec/i18n_spec.rb]: less than or equal?

Comment (adammw) [inline spec/i18n_spec.rb]: ha. lol

**Pull request merged**